File: src/main/java/com/github/aivancioglo/resttest/http/StatusCode.java
package com.github.aivancioglo.resttest.http;

/**
 * List of some main response codes that can be used in asserts.
 */
public enum StatusCode {
    OK(200, "OK"),
    CREATED(201, "Created"),
    NO_CONTENT(204, "No Content"),
    NOT_MODIFIED(304, "Not Modified"),
    BAD_REQUEST(400, "Bad Request"),
    UNAUTHORISE(401, "Unauthorised"),
    FORBIDDEN(403, "Forbidden"),
    NOT_FOUND(404, "Not Found"),
    CONFLICT(409, "Conflict"),
    INTERNAL_SERVER_ERROR(500, "Internal Server Error");

    public int code;
    public String message;

    StatusCode(int code, String message) {
        this.code = code;
        this.message = message;
    }
}

File: src/main/kotlin/com/github/aivancioglo/resttest/verifiers/StatusCode.kt
package com.github.aivancioglo.resttest.verifiers

import io.restassured.response.Response

/**
 * StatusCode verifier class.
 *
 * @constructor is creating variable of status code value.
 * @param statusCode of response.
 */
class StatusCode(private val statusCode: Int) : Verifier {

    /**
     * Verify response.
     *
     * @param response of your request.
     */
    override fun verify(response: Response) {
        response.then().statusCode(statusCode)
    }
}